capsules/extra/src/nonvolatile_storage_driver.rs

```diff
@@ -552,9 +1298,17 @@ impl SyscallDriver for NonvolatileStorage<'_> {
             0 => CommandReturn::success(),
 
             1 => {
-                // How many bytes are accessible from userspace
-                // TODO: Would break on 64-bit platforms
-                CommandReturn::success_u32(self.userspace_length as u32)
+                // How many bytes are accessible to each app
+                let res = self.apps.enter(processid, |app, _kernel_data| app.region);
+
+                // handle case where we fail to enter grant
+                res.map_or(CommandReturn::failure(ErrorCode::NOMEM), |region| {
+                    // handle case where app's region is not assigned
+                    region.map_or(CommandReturn::failure(ErrorCode::FAIL), |region| {
+                        // TODO: Would break on 64-bit platforms
```

Comment:
Why would this break on 64 bit platforms? Length can only be 24 bits right now, right?

Reply:
Yeah, I can remove this. 

This was here since before I made any changes. I just left it but now it's inaccurate.